packages/block-library/src/gallery/edit.js

```diff
@@ -366,10 +390,34 @@ function GalleryEdit( props ) {
 			const image = block.attributes.id
 				? imageData.find( ( { id } ) => id === block.attributes.id )
 				: null;
-			changedAttributes[ block.clientId ] = getHrefAndDestination(
-				image,
-				value
-			);
+
+			let hrefAndDestination = getHrefAndDestination( image, value );
+
+			if (
+				value === LINK_DESTINATION_LIGHTBOX &&
+				! lightboxSetting?.enabled
+			) {
+				hrefAndDestination = {
+					...hrefAndDestination,
+					lightbox: { enabled: true },
+				};
+			} else if (
+				value !== LINK_DESTINATION_NONE &&
+				value !== LINK_DESTINATION_LIGHTBOX &&
+				lightboxSetting?.enabled
+			) {
+				hrefAndDestination = {
+					...hrefAndDestination,
+					lightbox: { enabled: false },
+				};
+			} else {
+				hrefAndDestination = {
+					...hrefAndDestination,
+					lightbox: undefined,
+				};
+			}
```

Comment:
Is it possible to write this conditional statement inside the `hrefAndDestination` function?